src/ops.c

```diff
@@ -230,8 +230,8 @@ shift_line(
     int call_changed_bytes)	// call changed_bytes()
 {
     long long	count;
-    int		i, j;
-    int		sw_val = (int)get_sw_value_indent(curbuf);
+    long	i, j;
+    long	sw_val = get_sw_value_indent(curbuf);
```

Comment:
Changing `sw_val`, and the rest to `long` invalidates at least some of the overflow checking/calculation is this function (I think, probably depending on platform) for the case where `sizeof(long) == sizeof(long long)`

I wonder if something like the following would be useful (or exists?)
```
#define INT_TRIM(x) ((x) > INT_MAX ? INT_MAX : (x) < INT_MIN ? INT_MIN : (x))
```
Then could do
```
long sw_tmp = get_sw_value_indent(curbuf);
int  sw_val = INT_TRIM(sw_tmp);
```
In the code this PR adds to cindent.c could do
```
n = INT_TRIM(n);
```
right before the `switch`.

Using an inline function, rather than a macro, would be nice to avoid needing to manually create the tmp before using the macro.

Reply:
Let me merge it, seems to be okay now. Please go ahead and create a PR for this followup change then.

Reply:
> Let me merge it, seems to be okay now. Please go ahead and create a PR for this followup change then.

OK.

(I guess that "seems to be okay" means that options are now < INT_MAX)

Reply:
Yes, practically. I have not documented this explicitly, as I am not sure if we may need to support longs for other options specifically.